src/components/security_manager/src/security_manager_impl.cc

```diff
@@ -416,6 +416,20 @@ void SecurityManagerImpl::NotifyOnCertificateUpdateRequired() {
   }
 }
 
+void SecurityManagerImpl::NotifyListenersOnHandshakeFailed() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  std::list<SecurityManagerListener*>::iterator it = listeners_.begin();
+  while (it != listeners_.end()) {
```

Comment:
@AKalinich-Luxoft Use range based loop please

Reply:
@AByzhynar not sure that a good idea. I am using erase inside the loop, so it could break range iteration